apisix/core/config_etcd.lua

```diff
@@ -246,24 +248,9 @@ local function sync_data(self)
         return true
     end
 
-    -- for fetch the etcd index
-    local key_res, _ = getkey(self.etcd_cli, self.key)
-
     local dir_res, err = waitdir(self.etcd_cli, self.key, self.prev_index + 1, self.timeout)
-
     log.info("waitdir key: ", self.key, " prev_index: ", self.prev_index + 1)
     log.info("res: ", json.delay_encode(dir_res, true))
-    if err == "timeout" then
```

Comment:
why we delete this code block? it is useful for etcd `v2`. please confirm this

Reply:
Because etcd v3 does not have 1000 event history limit anymore, it could support unlimited history.

Reply:
But we also need to consider history compaction, or the accumulation of history would affect performance in a long term. Reference: https://etcd.io/docs/v3.4.0/op-guide/maintenance/#history-compaction

Reply:
@membphis  I think we don't need to consider compatibility with v2

Reply:
got it. many thanks for explain